### PR TITLE
Fix console logs page when linked from resource

### LINKED ISSUE #5764: [WebToolsE2E][Aspire] On the resources page of the dashboard, click Console logs under Actions. The Console page displays Initializing log viewer...
**REGRESSION INFO**: Worked in Aspire 8.2.0

**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install SDK 8.0.401
3. Install Aspire 9.0.0-preview.4.24467.9

   - Apply the dotnet8 feed for Aspire 9.0P4

**REPRO STEPS**

1. Open CMD, create an Aspire Starter App using following commands:
```
dotnet new aspire-starter -o AspireApp1
cd AspirerApp1/AspireApp1.AppHost
```
2. Dotnet run the project in CMD, and open URL in the browser.
`dotnet run`
3. On the resources page, click Console logs under Actions to jump to the Console page.

**NOTE**: If I click on 'Console' in the left navigation bar and select a resource, I can view the logs

**ACTUAL**
The Console page displays Initializing log viewer...
image

**EXPECTED**
On the resource page, click Console Log under Actions to see the logs
image

### PR DESCRIPTION
**Pull request by JamesNK**
I rewrote the console logs page [here](https://github.com/dotnet/aspire/pull/5703) but introduced a regression when loading logs when linked from a resource. Fixed in this PR with included test.

Fixes https://github.com/dotnet/aspire/issues/5764

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5776)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs [hunk 1]
```diff
@@ -57,15 +57,16 @@ private sealed class ConsoleLogsSubscription
     [Parameter]
     public string? ResourceName { get; set; }
 
+    private readonly TaskCompletionSource _logViewerReadyTcs = new();
```
Comment (davidfowl) [inline src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:60]: RunContinuationsAsynchronously

Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:60]: This is a case where that isn't desirable. The code after the log viewer is ready should run in the same context so it is on the UI thread.

Comment (davidfowl) [reply src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:60]: There's no UI thread. As long as the code is awaited the sync context will be preserved from the other side. The real question is whether the calling thread (the one calling Set result) wants to block inline while running the continuation.

Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs]: Blocking doesn't matter either way. I changed it.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**